### PR TITLE
feat(related_issues): Generate title and subtitle from events data

### PR DESCRIPTION
**Pull request by armenzg**
We need to fix the metadata for events to store the title and subtitle to match how issues are rendered in the product.

This change is a hack to speed up the related issues feature while we figure out the right way of fixing the data incosistency.

The screenshot that both the error event and the issue platform cards match the title/subtitle from the issue details page.
image

### CODE REVIEW

File: static/app/views/issueDetails/traceTimeline/traceIssue.tsx [hunk 1]
```diff
@@ -22,73 +18,63 @@ export function TraceIssueEvent({event}: TraceIssueEventProps) {
...
   const avatarSize = parseInt(space(4), 10);
 
   // If any of data fails to load, we don't want to render the component
   // Only "One other issue appears in the same trace. View Full Trace (X issues)" would show up
   return (
     <Fragment>
-      {isLoadingGroupData ? (
-        <LoadingIndicator mini />
-      ) : (
-        groupData && (
```
Comment (armenzg) [inline static/app/views/issueDetails/traceTimeline/traceIssue.tsx:48]: I'm removing these 4 lines. Hiding whitespaces makes it more obvious.
image

File: static/app/views/issueDetails/traceTimeline/traceIssue.tsx [hunk 2]
```diff
@@ -22,73 +18,63 @@ export function TraceIssueEvent({event}: TraceIssueEventProps) {
...
-                  shape="rect"
-                  width={`${projectBadgeSize}px`}
-                  height={`${projectBadgeSize}px`}
-                />
-              )}
-            </TraceIssueProjectBadge>
-            <TraceIssueDetailsContainer>
-              <NoOverflowDiv>
-                <TraceIssueEventTitle>
-                  {groupData.metadata.title || groupData.metadata.type}
```
Comment (armenzg) [inline static/app/views/issueDetails/traceTimeline/traceIssue.tsx:77]: I change this for `{title}`.

File: static/app/views/issueDetails/traceTimeline/traceIssue.tsx [hunk 3]
```diff
@@ -22,73 +18,63 @@ export function TraceIssueEvent({event}: TraceIssueEventProps) {
...
-                </TraceIssueEventTitle>
-                <TraceIssueEventTransaction>
-                  {event.transaction}
-                </TraceIssueEventTransaction>
-              </NoOverflowDiv>
-              <NoOverflowDiv>{groupData.metadata.value}</NoOverflowDiv>
```
Comment (armenzg) [inline static/app/views/issueDetails/traceTimeline/traceIssue.tsx:83]: I change this for `{subtitle}`.

**Comment by vartec** [comment]
is there a backend component for this?

**Comment by armenzg** [comment]
> is there a backend component for this?

No backend component is required. In this PR, I add the `message` field for the `events/` endpoint call.

File: static/app/views/issueDetails/traceTimeline/traceIssue.tsx [hunk 4]
```diff
@@ -22,73 +18,63 @@ export function TraceIssueEvent({event}: TraceIssueEventProps) {
...
+      </TraceIssueLinkContainer>
     </Fragment>
   );
 }
 
+function getTitleSubtitle(event: TimelineEvent) {
+  let title;
+  let subtitle;
+  if (event['event.type'] === 'error') {
+    title = event.title.split(':')[0];
```
Comment (malwilley) [inline static/app/views/issueDetails/traceTimeline/traceIssue.tsx:69]: I'm okay with this as a temporary hack, but if we go forward with this I'd want to make sure that this matches the logic we have in the backend

**Review by malwilley — APPROVED** [review]

File: static/app/views/issueDetails/traceTimeline/traceIssue.tsx (hunk 4 above)
Comment (armenzg) [reply static/app/views/issueDetails/traceTimeline/traceIssue.tsx:69]: It's on my list of tasks.

**Pull request merged**